tikv/raftstore/peer_storage.go

```diff
@@ -134,6 +151,32 @@ func (ec *EntryCache) compactTo(idx uint64) {
 	ec.cache = ec.cache[pos:]
 }
 
+type ApplySnapResult struct {
+	// PrevRegion is the region before snapshot applied
+	PrevRegion *metapb.Region
+	Region     *metapb.Region
+}
+
+type RegionTaskType int
+
+const (
+	RegionTaskType_Gen RegionTaskType = 0 + iota
+	RegionTaskType_Apply
+	/// Destroy data between [start_key, end_key).
+	///
+	/// The deletion may and may not succeed.
+	RegionTaskType_Destroy
+)
+
+type RegionTask struct {
+	RegionId uint64
+	TaskType RegionTaskType
+	Notifier chan<- eraftpb.Snapshot
+	Status   *JobStatus
```

Comment:
We don't need to use pointer.

Reply:
In TiKV, it is a atomic usize shared by multiple threads.

Reply:
I know, we can use int64 type to do atomic operations.

Reply:
Yes, JobStatus is already int/int64.

Reply:
And `JobStatus` shared by `RegionTask` and `SnapState` which in different threads.